swift-generator/src/main/java/com/facebook/swift/generator/template/StructContext.java

```diff
@@ -51,6 +52,26 @@ public boolean getHasUniqueFieldTypes()
       return fieldTypesSet.size() == fieldTypesList.size();
     }
 
+    public boolean getHasUniqueFieldTypesUnderErasure()
+    {
+        List<String> fieldTypesList = Lists.transform(this.fields, new Function<FieldContext, String>() {
+            public String apply(FieldContext field) {
+                // remove generics
+                String javaType = field.getJavaType();
+                int startTypeParameter = javaType.indexOf('<');
```

Comment:
I wonder if < is a valid character in a Java class (not that people should be using that anyways).